README.md
# Astronomic Calendar

## What is _Astronomic Calendar_?

The goal is to provide a script that prompts a calendar which takes different options to adapt to a more modern way of life and away from religious _dogmas_.

The project aims to create a _Universal_, _Multicultural_ and _Secular_ calendar that can be used __Worldwide__.

## Funcionamiento

The __Astronomic Calendar__ is organized in _years_, _months_, _weeks_ and _days_.

A __year__ is a full cicle of the _Earth_ arround the _Sun_. The actual difference is that leap years are not preset in regular cicles, but for the actual date of the _equinox_ of March.

__Months__ are defined by the phases of the _Moon_. So, for every _Moon_ __phases__ cicle we have a month. It is still to be determined if months, as well as years, are defined by the actual phases or they are prestablished in regular cicles.
Since there is not a predetermined fixed number of _lunar phases_ for every year, months will vary between 12 and 13 per year.

__Weeks__ have 5 days each. Therefore, every year has 73 _weeks_.
This determines, as well, that every year begins with a new week. In the case of _leap years_ a day is added to the last week of the year as a __worldwide festivity__.

## Actual progress

This is a very early stage so don't expect much yet.

## Collaborators

Always open to new ideas and collaborations.


CHANGELOG.md
# CHANGELOG: Astronomic Calendar
## 0.2.2 - 2020/10/10
 ### Added
  + New data format in './js/astroData.js'/
 ### Fixed
  + Shows New Year
 ### Issues
  + A lot of old code not being used
## 0.2.1 - 2020/10/05
 ### Changed
  - Astronomic data from './js/data.js' now loaded from './js/AData.js'
  - Load _ACalendar Class_ & _Astronomic Data_ to module './js/index.js'
## 0.1.5 - 2020/10/05
 ### Fixed
  - gregorian calendar working again
## 0.1.4 - 2020/10/05
 ### Added
  - new CSS selectors
  - empty days now different format
### Fixed
 - Display different calendars at the same time whitin its containers
 - FirstDayOfMonth issues
### Changed
 - printScreen() => printMonth()
## 0.1.3 - 2020/10/04
 ### Added
  - new class method: printCalendar()
  - first day of month is next day of last week of month before
## 0.1.2
 ### Added
 - Added month name
## 0.1.1
 ### Fixed
 - All CSS in index.css
 - Some CSS fixes
 ### Removed
 - Deleted lines for 'test' and 'json' in index.html
 - Deleted some commented lines in dateClass.js
 ### Changed
 - Few changes in Setters & Getters
## 0.1.0

js/index.js
import AData from './AData.js'
import astroData from './astroData.js'
import { ACalendar } from './ACalendar.js';

/**
 * ASTRONOMIC CALENDAR NEW OPTIONS
 */

// Array of day names
const weekDaysARRAY = ["venus", "marte", "júpiter", "sol", "luna"];

// Array of month names
const monthsARRAY = [
  "capricornio",
  "acuario",
  "piscis",
  "aries",
  "tauro",
  "géminis",
  "cáncer",
  "leo",
  "virgo",
  "libra",
  "escorpio",
  "sagitario",
  "andrómeda",
];

/**
 * CALENDARS
 */

const calendar = new ACalendar(2018, 2019, AData, astroData );

console.log("setfullmoon: ", (calendar.fullMoon = true));

let test = calendar.astroData
console.log({test})
calendar.includeSunPhase = "MaxLatSouth";
calendar.addSunData = 'equiNorth'
let arrSun = calendar.sunData
console.log({arrSun})
//calendar.printMonth(1, 12, "2019, 4, 5");

calendar.weekDays = weekDaysARRAY;
calendar.monthNames = monthsARRAY;

calendar.printCalendar(2, 3, [2020, 2, 18], 3);

js/classes.js
/**
 * AMonth() Class
 * It is initialized with the whole year data
 * Then we can call any month of the year to be printed
 */

class AMonth {
    constructor(firstDayOfMonth, firstGregDate, monthName, days, weekDays, astroData) {
        this._firstDayOfMonth = firstDayOfMonth;
        this._firstGregDate = firstGregDate;
        this._monthName = monthName;
        this._days = days;
        this._weekDays = weekDays;
        this._astroData = astroData;
    }
    moon() {
        return this._astroData['moon']
    }
    sun() {
        return this._astroData['sun']
    }
    /**
     * returns a months data
     * @param [0-12] n 
     */
    getMonth(n) {
        return {
            name: this._monthName,
            days: this._days,
            weeks: [
                [1, [0, 0, 1, 2, 3]],
                [2, [4, 5, 6, 7, 8]],
                [3, [9, 10, 11, 12, 13]],
                [4, [14, 15, 16, 17, 18]],
                [5, [19, 20, 21, 22, 23]],
                [6, [24, 25, 26, 27, 28]],
                [7, [29, 30]]
            ],
            moon: this.moon(),
            sun: this.sun()
        }
    }
}

export { AMonth }

js/ACalendar.js
class ACalendar {
  constructor(yearZero, gregYear, data, astroData) {
    this._yearZero = yearZero;
    this._year = gregYear - yearZero;
    this._gregYear = gregYear;
    this._data = data;
    this._astroData = astroData;
    this._sunData = [];
  }
  /**
   * DEFAULT VALUES
   */
  _weekDays = [
    "lunes",
    "martes",
    "miércoles",
    "jueves",
    "viernes",
    "sábado",
    "domingo",
  ];
  _months = [
    "enero",
    "febrero",
    "marzo",
    "abril",
    "mayo",
    "junio",
    "julio",
    "agosto",
    "septiembre",
    "octubre",
    "noviembre",
    "diciembre",
  ];
  /**
   * MODIFY DEFAULT VALUES
   * SETTERS & GETTERS
   */
  set weekDays(arr) {
    this._weekDays = arr;
    console.log("weekDays = ", arr);
  }
  /**
   * @param {string[]} arr
   */
  set monthNames(arr) {
    this._months = arr;
    console.log("months = ", arr);
  }
  /**
   * @param {String} sunPhase
   */
  set includeSunPhase(sunPhase) {
    let arr = [];
    this._data["sun"].forEach((s) => {
      if (s.phase == sunPhase) {
        console.log(s.date);
        //arr.push(this.toLocalDate(s.date));
        arr.push(new Date(s.date).toLocaleDateString());
      }
    });
    console.log("Sun Phases set: ", this._data, arr);
    this._sunData = arr;
  }
  /**
   * @param {string} sunPhase
   * get sunPhase for before, actual and next year
   */
  set addSunData(sunPhase) {
    let year = this._gregYear -1
    for (let i = 0; i <= 2; i++) {
      let sunDate =this._astroData['sun'][year + i][sunPhase]['gregDate']
      this._sunData.push((new Date(sunDate).toLocaleDateString()))
    }
  }
  // yearZero
  set yearZero(y) {
    this._yearZero = y;
  }
  get yearZero() {
    return this._yearZero;
  }
  // Astronomic Data
  get data() {
    return this._data;
  }
  get astroData() {
    return this._astroData
  }
  get sunData() {
    return this._sunData
  }
  // Leap year
  conf_leapYears = true;
  get leapYears() {
    return this.conf_leapYears;
  }
  set leapYears(x) {
    this.conf_leapYears = x;
    console.log("leapYears = ", x);
  }
  // Show full moon
  showFullMoon = true;
  get fullMoon() {
    return this.showFullMoon;
  }
  set fullMoon(x) {
    this.showFullMoon = x;
    console.log("showFullMoon = ", x);
  }
  // Show new Moon
  showNewMoon = true;
  get newMoon() {
    return this.showNewMoon;
  }
  set newMoon(x) {
    this.showNewMoon = x;
  }
  /** CONFIGURATION */
  // locale
  conf_locale = "es-ES";
  set locale(x) {
    this.conf_locale = x;
    console.log("conf_locale: ", locale);
  }
  get locale() {
    return this.conf_locale;
  }
  // Moon cicle
  conf_moonCicle = false;
  /**
   * @param {boolean} x
   */
  set moonCicle(x) {
    this.conf_moonCicle = x;
    console.log("conf_moonCicle: ", this.moonCicle);
  }
  get conf_moonCicle() {
    return this.conf_moonCicle;
  }

  /** INTERNAL FUNCTIONS */
  // DATES
  toLocalDate(d) {
    let day = new Date(d);
    let localeDate = `${day.toLocaleString(this.conf_locale)}`;
    console.log({ localeDate });
    return localeDate.split(" ");
  }
  addDays(date, days) {
    var result = new Date(date);
    result.setDate(result.getDate() + days);
    return result;
  }

  // SUN
  // MOON
  fullMoonDates(data) {
    let arr = [];
    data.forEach((l) => {
      if (l.phase == "luna llena") {
        arr.push(l.date.NSC.toLocaleDateString());
      }
    });
    return arr;
  }
  newMoonDates(data) {
    let arr = [];
    data.forEach((l) => {
      if (l.phase == "luna nueva") {
        arr.push(l.date.NSC.toLocaleDateString());
      }
    });
    return arr;
  }
  moonPhaseDates(data, mPhase) {
    let arr = [];
    data.forEach((l) => {
      if (l.phase == mPhase) {
        arr.push(l.date.NSC.toLocaleDateString());
      }
    });
    return arr;
  }
  /** GETTERS */
  get numWeekDays() {
    return this._weekDays.length;
  }
  get weekDays() {
    return this._weekDays;
  }

  /** CALENDAR */
  get monthsDays() {
    const monthDays = this._months.map((x, i) => {
      return i % 2 === 0 ? 30 : 29;
    });
    if (this.conf_leapYears && this._year % 2 == 0) monthDays[3] = 30;
    if (this.conf_moonCicle && this._year % 33 === 0) monthDays[1] = 30;
    return monthDays;
  }
  // m = NSC month index
  // date = first day of de month in gregorian time
  // in original formula is also: moon = moon data
  printMonthDays(firstDayOfMonth, m, date) {
    let html = "",
      printWeekResp,
      lastDayOfMonth;
    let days = this.monthsDays[m];
    let numberOfWeeks = Math.ceil(
      (this.monthsDays[m] + firstDayOfMonth) / this.numWeekDays
    );
    let firstDayDate = date;
    let firstDayOfWeek = firstDayOfMonth;
    for (let r = 1; r <= numberOfWeeks; ++r) {
      printWeekResp = this.printWeek(r, days, firstDayOfMonth, firstDayDate);
      html += `<div class="week week-${r}"><div class="week-number">${r}</div>`;
      html += printWeekResp.html;
      html += `</div>`;
      firstDayDate = this.addDays(
        firstDayDate,
        this.numWeekDays - firstDayOfWeek
      );
      firstDayOfWeek = 0;
      if (r == numberOfWeeks) {
        lastDayOfMonth = printWeekResp.lastDayOfWeek;
      }
    }
    return { html, lastDayOfMonth };
  }
  sunClass(d) {
    return this._sunData.includes(d) ? "new-year" : "void";
  }
  // w = week of the month
  // days = number of days in the week
  // firstDay = first day of the week
  // date = gregorian date of first day of the week.
  printWeek(w, days, firstDay, date) {
    let html = "",
      newDate,
      lastDayOfWeek = 0;
    let d = w * this.numWeekDays - (this.numWeekDays + firstDay) + 1;
    for (let c = 1; c <= this.numWeekDays; ++c) {
      newDate = date.toLocaleDateString();
      if (firstDay >= c && w == 1) {
        html += `<div class="day"></div>`;
        d++;
      } else if (d <= days) {
        html += `<div class="day day-${c} ${this.sunClass(
          newDate
        )}" title="${newDate}">
                <span class="day-number">${d}</span>
                </div>`;
        date = this.addDays(date, 1);
        d++;
      } else {
        if (lastDayOfWeek == 0) lastDayOfWeek = c - 1;
        html += `<div class="day"></div>`;
      }
    }
    //if (lastDayOfMonth == undefined) lastDayOfWeek = 0
    return { html, lastDayOfWeek };
  }
  printDayNames() {
    let dayNames = '<div class="day-names"><div class="week-number"></div>';
    for (let d = 0; d < this.numWeekDays; d++) {
      dayNames += `<div class="day-name day day-${d + 1}">${
        this._weekDays[d]
      }</div>`;
    }
    dayNames += "</div>";
    return dayNames;
  }
  printMonthName(m) {
    let monthName = this._months[m - 1].replace(/^\w/, (c) => c.toUpperCase());
    return monthName;
  }
  printMonth(firstDay, month, date) {
    const elMonthName = document.querySelector("#month-name");
    elMonthName.innerHTML = `<div class="month-name">${this.printMonthName(month)}</div>`;

    const elDayNames = document.querySelector("#day-names");
    elDayNames.innerHTML = this.printDayNames();

    const elMonthDays = document.querySelector("#month-days");
    elMonthDays.innerHTML = (this.printMonthDays(
      firstDay,
      month - 1,
      new Date(date)
    )).html
  }
  /**
   * printCalendar()
   * @param {Number} firstDay / Day of the week of first day of first month /
   * @param {Number} month / first month of astronomic calendar to print /
   * @param {Array} date / array of first day correspondant in Gregorian Calendar [year, month, day] /
   * @param {Number} n / number of months to print / 
   */
  printCalendar(firstDay, month, date, n) {
    let calendarString = "";
    let days, monthDays, firstDayOfMonth;
    let gregDate = new Date(date);

    const dayNames = this.printDayNames();

    for (let i = 0; i < n; i++) {
      days = this.monthsDays[month - 1 + i];
      firstDayOfMonth =
        firstDayOfMonth == undefined ? firstDay : monthDays.lastDayOfMonth;

      console.log({ firstDayOfMonth });

      monthDays = this.printMonthDays(firstDayOfMonth, month - 1 + i, gregDate);

      calendarString += `<div class="month-name">${this.printMonthName(month + i)}</div>`;
      calendarString += dayNames;
      calendarString += monthDays.html;

      console.log({ gregDate, days });
      gregDate = this.addDays(gregDate, days);
    }
    const elMonthName = document.querySelector("#calendar");
    elMonthName.innerHTML = calendarString;
  }
printCalendarYear(y) {
  let calendarString = "";
  let days, monthDays, firstDayOfMonth;
  let gregDate = new Date(date);

  const dayNames = this.printDayNames();

  for (let i = 0; i < n; i++) {
    days = this.monthsDays[month - 1 + i];
    firstDayOfMonth =
      firstDayOfMonth == undefined ? firstDay : monthDays.lastDayOfMonth;

    console.log({ firstDayOfMonth });

    monthDays = this.printMonthDays(firstDayOfMonth, month - 1 + i, gregDate);

    calendarString += `<div class="month-name">${this.printMonthName(month + i)}</div>`;
    calendarString += dayNames;
    calendarString += monthDays.html;

    console.log({ gregDate, days });
    gregDate = this.addDays(gregDate, days);
  }
  const elMonthName = document.querySelector("#calendar");
  elMonthName.innerHTML = calendarString;
}
}
export { ACalendar}

js/data.js
const astronomicData = {
  sources: {
      sun: [
            "http://www.vaxasoftware.com/doc_edu/ast/estaciones_tierra.pdf"
      ],
      moon: [
          "https://www.calendario-365.es/luna/lunar-fases.html",
          "https://canaltrece.com.co/noticias/facebook-messenger-instagram-whatsapp-novedades/"
      ]
  },
  sun: [
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2019-12-22T04:19:00.000Z",
    },
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2020-12-21T10:02:00.000Z",
    },
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2021-12-21T15:59:00.000Z",
    },
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2022-12-21T21:48:00.000Z",
    },
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2023-12-22T03:27:00.000Z",
    },
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2024-12-21T09:21:00.000Z",
    },
    {
      distance: "",
      phase: "MaxLatSouth",
      date: "2025-12-21T15:03:00.000Z",
    },
  ],
  moon: [
    {
      "dist": 404404,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1578030412000, "greg": "2020-01-03T05:46:52.000Z" }
    },
    {
      "dist": 375375,
      "phase": "Luna llena",
      "date": { "NSC": 1578687794000, "greg": "2020-01-10T20:23:14.000Z" }
    },
    {
      "dist": 365365,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1579269635000, "greg": "2020-01-17T14:00:35.000Z" }
    },
    {
      "dist": 395395,
      "phase": "Luna nueva",
      "date": { "NSC": 1579905851000, "greg": "2020-01-24T22:44:11.000Z" }
    },
    {
      "dist": 399399,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1580611408000, "greg": "2020-02-02T02:43:28.000Z" }
    },
    {
      "dist": 368368,
      "phase": "Luna llena",
      "date": { "NSC": 1581237275000, "greg": "2020-02-09T08:34:35.000Z" }
    },
    {
      "dist": 371371,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1581808749000, "greg": "2020-02-15T23:19:09.000Z" }
    },
    {
      "dist": 402402,
      "phase": "Luna nueva",
      "date": { "NSC": 1582475624000, "greg": "2020-02-23T16:33:44.000Z" }
    },
    {
      "dist": 391391,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1583182718000, "greg": "2020-03-02T20:58:38.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Luna llena",
      "date": { "NSC": 1583779715000, "greg": "2020-03-09T18:48:35.000Z" }
    },
    {
      "dist": 379379,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1584354939000, "greg": "2020-03-16T10:35:39.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Luna nueva",
      "date": { "NSC": 1585045776000, "greg": "2020-03-24T10:29:36.000Z" }
    },
    {
      "dist": 381381,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1585743712000, "greg": "2020-04-01T12:21:52.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Luna llena",
      "date": { "NSC": 1586320536000, "greg": "2020-04-08T04:35:36.000Z" }
    },
    {
      "dist": 389389,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1586912217000, "greg": "2020-04-15T00:56:57.000Z" }
    },
    {
      "dist": 403403,
      "phase": "Luna nueva",
      "date": { "NSC": 1587616020000, "greg": "2020-04-23T04:27:00.000Z" }
    },
    {
      "dist": 372372,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1588286324000, "greg": "2020-04-30T22:38:44.000Z" }
    },
    {
      "dist": 366366,
      "phase": "Luna llena",
      "date": { "NSC": 1588855535000, "greg": "2020-05-07T12:45:35.000Z" }
    },
    {
      "dist": 397397,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1589472198000, "greg": "2020-05-14T16:03:18.000Z" }
    },
    {
      "dist": 397397,
      "phase": "Luna nueva",
      "date": { "NSC": 1590176389000, "greg": "2020-05-22T19:39:49.000Z" }
    },
    {
      "dist": 366366,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1590816622000, "greg": "2020-05-30T05:30:22.000Z" }
    },
    {
      "dist": 373373,
      "phase": "Luna llena",
      "date": { "NSC": 1591391563000, "greg": "2020-06-05T21:12:43.000Z" }
    },
    {
      "dist": 403403,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1592036699000, "greg": "2020-06-13T08:24:59.000Z" }
    },
    {
      "dist": 389389,
      "phase": "Luna nueva",
      "date": { "NSC": 1592728936000, "greg": "2020-06-21T08:42:16.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1593339390000, "greg": "2020-06-28T10:16:30.000Z" }
    },
    {
      "dist": 381381,
      "phase": "Luna llena",
      "date": { "NSC": 1593931497000, "greg": "2020-07-05T06:44:57.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1594603888000, "greg": "2020-07-13T01:31:28.000Z" }
    },
    {
      "dist": 379379,
      "phase": "Luna nueva",
      "date": { "NSC": 1595273616000, "greg": "2020-07-20T19:33:36.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1595860427000, "greg": "2020-07-27T14:33:47.000Z" }
    },
    {
      "dist": 390390,
      "phase": "Luna llena",
      "date": { "NSC": 1596477578000, "greg": "2020-08-03T17:59:38.000Z" }
    },
    {
      "dist": 402402,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1597171668000, "greg": "2020-08-11T18:47:48.000Z" }
    },
    {
      "dist": 371371,
      "phase": "Luna nueva",
      "date": { "NSC": 1597812135000, "greg": "2020-08-19T04:42:15.000Z" }
    },
    {
      "dist": 367367,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1598385551000, "greg": "2020-08-25T19:59:11.000Z" }
    },
    {
      "dist": 399399,
      "phase": "Luna llena",
      "date": { "NSC": 1599031392000, "greg": "2020-09-02T07:23:12.000Z" }
    },
    {
      "dist": 396396,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1599737298000, "greg": "2020-09-10T11:28:18.000Z" }
    },
    {
      "dist": 365365,
      "phase": "Luna nueva",
      "date": { "NSC": 1600347659000, "greg": "2020-09-17T13:00:59.000Z" }
    },
    {
      "dist": 374374,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1600919794000, "greg": "2020-09-24T03:56:34.000Z" }
    },
    {
      "dist": 404404,
      "phase": "Luna llena",
      "date": { "NSC": 1601593615000, "greg": "2020-10-01T23:06:55.000Z" }
    },
    {
      "dist": 387387,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1602297684000, "greg": "2020-10-10T02:41:24.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Luna nueva",
      "date": { "NSC": 1602883934000, "greg": "2020-10-16T21:32:14.000Z" }
    },
    {
      "dist": 383383,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1603466657000, "greg": "2020-10-23T15:24:17.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Luna llena",
      "date": { "NSC": 1604159489000, "greg": "2020-10-31T15:51:29.000Z" }
    },
    {
      "dist": 377377,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1604846858000, "greg": "2020-11-08T14:47:38.000Z" }
    },
    {
      "dist": 364364,
      "phase": "Luna nueva",
      "date": { "NSC": 1605420534000, "greg": "2020-11-15T06:08:54.000Z" }
    },
    {
      "dist": 393393,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1606023954000, "greg": "2020-11-22T05:45:54.000Z" }
    },
    {
      "dist": 401401,
      "phase": "Luna llena",
      "date": { "NSC": 1606732324000, "greg": "2020-11-30T10:32:04.000Z" }
    },
    {
      "dist": 369369,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1607391479000, "greg": "2020-12-08T01:37:59.000Z" }
    },
    {
      "dist": 369369,
      "phase": "Luna nueva",
      "date": { "NSC": 1607966325000, "greg": "2020-12-14T17:18:45.000Z" }
    },
    {
      "dist": 401401,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1608597751000, "greg": "2020-12-22T00:42:31.000Z" }
    },
    {
      "dist": 393393,
      "phase": "Luna llena",
      "date": { "NSC": 1609302624000, "greg": "2020-12-30T04:30:24.000Z" }
    },
    {
      "dist": 364364,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1609929515000, "greg": "2021-01-06T10:38:35.000Z" }
    },
    {
      "dist": 377377,
      "phase": "Luna nueva",
      "date": { "NSC": 1610517757000, "greg": "2021-01-13T06:02:37.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1611180215000, "greg": "2021-01-20T22:03:35.000Z" }
    },
    {
      "dist": 383383,
      "phase": "Luna llena",
      "date": { "NSC": 1611865115000, "greg": "2021-01-28T20:18:35.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1612463922000, "greg": "2021-02-04T18:38:42.000Z" }
    },
    {
      "dist": 387387,
      "phase": "Luna nueva",
      "date": { "NSC": 1613074091000, "greg": "2021-02-11T20:08:11.000Z" }
    },
    {
      "dist": 404404,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1613764146000, "greg": "2021-02-19T19:49:06.000Z" }
    },
    {
      "dist": 374374,
      "phase": "Luna llena",
      "date": { "NSC": 1614417576000, "greg": "2021-02-27T09:19:36.000Z" }
    },
    {
      "dist": 365365,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1614997920000, "greg": "2021-03-06T02:32:00.000Z" }
    },
    {
      "dist": 396396,
      "phase": "Luna nueva",
      "date": { "NSC": 1615634612000, "greg": "2021-03-13T11:23:32.000Z" }
    },
    {
      "dist": 398398,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1616341306000, "greg": "2021-03-21T15:41:46.000Z" }
    },
    {
      "dist": 367367,
      "phase": "Luna llena",
      "date": { "NSC": 1616964604000, "greg": "2021-03-28T20:50:04.000Z" }
    },
    {
      "dist": 371371,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1617537852000, "greg": "2021-04-04T12:04:12.000Z" }
    },
    {
      "dist": 402402,
      "phase": "Luna nueva",
      "date": { "NSC": 1618201976000, "greg": "2021-04-12T04:32:56.000Z" }
    },
    {
      "dist": 390390,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1618909201000, "greg": "2021-04-20T09:00:01.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Luna llena",
      "date": { "NSC": 1619501584000, "greg": "2021-04-27T05:33:04.000Z" }
    },
    {
      "dist": 380380,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1620078703000, "greg": "2021-05-03T21:51:43.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Luna nueva",
      "date": { "NSC": 1620766893000, "greg": "2021-05-11T21:01:33.000Z" }
    },
    {
      "dist": 381381,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1621458793000, "greg": "2021-05-19T21:13:13.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Luna llena",
      "date": { "NSC": 1622034891000, "greg": "2021-05-26T13:14:51.000Z" }
    },
    {
      "dist": 389389,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1622625964000, "greg": "2021-06-02T09:26:04.000Z" }
    },
    {
      "dist": 403403,
      "phase": "Luna nueva",
      "date": { "NSC": 1623329645000, "greg": "2021-06-10T12:54:05.000Z" }
    },
    {
      "dist": 372372,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1623995684000, "greg": "2021-06-18T05:54:44.000Z" }
    },
    {
      "dist": 366366,
      "phase": "Luna llena",
      "date": { "NSC": 1624567214000, "greg": "2021-06-24T20:40:14.000Z" }
    },
    {
      "dist": 397397,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1625181159000, "greg": "2021-07-01T23:12:39.000Z" }
    },
    {
      "dist": 397397,
      "phase": "Luna nueva",
      "date": { "NSC": 1625887063000, "greg": "2021-07-10T03:17:43.000Z" }
    },
    {
      "dist": 366366,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1626523897000, "greg": "2021-07-17T12:11:37.000Z" }
    },
    {
      "dist": 373373,
      "phase": "Luna llena",
      "date": { "NSC": 1627101447000, "greg": "2021-07-24T04:37:27.000Z" }
    },
    {
      "dist": 403403,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1627744696000, "greg": "2021-07-31T15:18:16.000Z" }
    },
    {
      "dist": 389389,
      "phase": "Luna nueva",
      "date": { "NSC": 1628437846000, "greg": "2021-08-08T15:50:46.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1629048064000, "greg": "2021-08-15T17:21:04.000Z" }
    },
    {
      "dist": 381381,
      "phase": "Luna llena",
      "date": { "NSC": 1629640935000, "greg": "2021-08-22T14:02:15.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1630314902000, "greg": "2021-08-30T09:15:02.000Z" }
    },
    {
      "dist": 379379,
      "phase": "Luna nueva",
      "date": { "NSC": 1630983121000, "greg": "2021-09-07T02:52:01.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1631572880000, "greg": "2021-09-13T22:41:20.000Z" }
    },
    {
      "dist": 391391,
      "phase": "Luna llena",
      "date": { "NSC": 1632189284000, "greg": "2021-09-21T01:54:44.000Z" }
    },
    {
      "dist": 402402,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1632887904000, "greg": "2021-09-29T03:58:24.000Z" }
    },
    {
      "dist": 371371,
      "phase": "Luna nueva",
      "date": { "NSC": 1633525544000, "greg": "2021-10-06T13:05:44.000Z" }
    },
    {
      "dist": 367367,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1634102855000, "greg": "2021-10-13T05:27:35.000Z" }
    },
    {
      "dist": 399399,
      "phase": "Luna llena",
      "date": { "NSC": 1634749061000, "greg": "2021-10-20T16:57:41.000Z" }
    },
    {
      "dist": 395395,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1635458804000, "greg": "2021-10-28T22:06:44.000Z" }
    },
    {
      "dist": 365365,
      "phase": "Luna nueva",
      "date": { "NSC": 1636064126000, "greg": "2021-11-04T22:15:26.000Z" }
    },
    {
      "dist": 375375,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1636638502000, "greg": "2021-11-11T13:48:22.000Z" }
    },
    {
      "dist": 404404,
      "phase": "Luna llena",
      "date": { "NSC": 1637315981000, "greg": "2021-11-19T09:59:41.000Z" }
    },
    {
      "dist": 386386,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1638019791000, "greg": "2021-11-27T13:29:51.000Z" }
    },
    {
      "dist": 363363,
      "phase": "Luna nueva",
      "date": { "NSC": 1638607470000, "greg": "2021-12-04T08:44:30.000Z" }
    },
    {
      "dist": 384384,
      "phase": "Cuarto creciente",
      "date": { "NSC": 1639190252000, "greg": "2021-12-11T02:37:32.000Z" }
    },
    {
      "dist": 405405,
      "phase": "Luna llena",
      "date": { "NSC": 1639892278000, "greg": "2021-12-19T05:37:58.000Z" }
    },
    {
      "dist": 376376,
      "phase": "Cuarto menguante",
      "date": { "NSC": 1640575560000, "greg": "2021-12-27T03:26:00.000Z" }
    }
  ]  
};
